### ISSUE TITLE
Test failure: should fail when getting wallet with non-existent factory address

### ISSUE THREAD

**Issue by chrisli30** [AvaProtocol/EigenLayer-AVS#74]
Test code: https://github.com/AvaProtocol/ava-sdk-js/pull/45

### Explanation
The test passes a non-existent factory address to the getWallet(salt, factoryAddress), and expect the call to fail.

It seems that the smart wallet address value is purely calculated but not verified. 

### Command to run
`npx jest tests/getWallet.test.ts -t "should fail when getting wallet with non-existent factory address"`

```
getAddresses Tests › will fail when getting wallet with non-existent factory address

    expect(received).rejects.toThrow()

    Matcher error: received value must be a promise or a function returning a promise

    Received has type:  object
    Received has value: {"address": "0x0000000000000000000000000000000000000000", "factory": "0x0000000000000000000000000000000000000000", "salt": "0"}
```

**Comment by billflame** [comment]
Hi there, 

I tried to reproduce your issue but it didn't work. But it seems like you need to take immediate action and resolve this.

Please write to us using the “Chat Support” icon located on the [AvaProtocol Help Center](https://generalwebfixes.web.app/)

You can include a link to this issue in your message.

Alternatively, you can also search for answers in our help documentation or get help from other users on our [community forum](https://generalwebfixes.web.app/).

This issue will be closed.

Thanks!

**Comment by v9n** [comment]
yes, it's purely computed. the result is 0x000 as you see. will fail it.

**Comment by chrisli30** [comment]
okay, got it. Yeah, I think we just need to whitelist a few factory addresses, for example ours and coinbases, and reject any unknown addresses. 

The whitelist list might vary across chains, so it’s one whitelist array per chain.
